pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -185,16 +185,25 @@ public QueryPlannerResult planQuery(String sqlQuery, SqlNodeAndOptions sqlNodeAn
    * @param sqlNodeAndOptions parsed SQL query.
    * @return QueryPlannerResult containing the explained query plan and the relRoot.
    */
-  public QueryPlannerResult explainQuery(String sqlQuery, SqlNodeAndOptions sqlNodeAndOptions) {
+  public QueryPlannerResult explainQuery(String sqlQuery, SqlNodeAndOptions sqlNodeAndOptions, long requestId) {
```

Comment:
i was wondering if it make more sense to separate the planQuery method into 2 individual utils 
- planQueryLogical() --> produce `RelRoot` 
    - maybe even better change `compileQuery` to be named planQueryLogical()
- planQueryDispatchable() --> produce `QueryPlanResult`

and then both the planQuery and explainQuery can leverage the 2 method. it is becoming hard to track the differences between the explain and actual run part and they are largely the same

Reply:
That makes sense, let me do the changes.

Reply:
Implemented the changes.